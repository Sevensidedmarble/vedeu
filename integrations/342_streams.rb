#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

TESTCASE = '342_streams'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    height 10
    debug!
    log '/tmp/vedeu_views_dsl.log'
    renderers(Vedeu::Renderers::Terminal.new(
                filename: "/tmp/#{TESTCASE}.out",
                write_file: true))
    run_once!
    standalone!
  end

  Vedeu.interface :test1_interface do
    background '#000022'
    geometry do
      x  3
      y  3
      xn 53
      yn 3
    end
  end

  Vedeu.render do
    view(:test1_interface) do
      line do
        stream do
          foreground '#cc66ff'
          left '07/01/2016 21:32:19'
        end
        stream do
          foreground '#007777'
          left ' null'
        end
        stream do
          foreground '#dddddd'
          left ' > hi'
        end
      end
    end
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

DSLApp.start

load File.dirname(__FILE__) + '/test_runner.rb'
TestRunner.result(TESTCASE, __FILE__)
