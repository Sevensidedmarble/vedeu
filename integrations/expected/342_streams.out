[0m[3;3H[38;5;177m[48;5;16m0[3;4H7[3;5H/[3;6H0[3;7H1[3;8H/[3;9H2[3;10H0[3;11H1[3;12H6[3;13H [3;14H2[3;15H1[3;16H:[3;17H3[3;18H2[3;19H:[3;20H1[3;21H9[3;22H[38;5;30m[48;5;16m [3;23Hn[3;24Hu[3;25Hl[3;26Hl[3;27H[38;5;188m[48;5;16m [3;28H>[3;29H [3;30Hh[3;31Hi[0m